File: Matlab/read_ptv_is_files.m
function data = read_ptv_is_files(directory,first,last)
% to be added - READ_PTV_IS_FILES(DIRECTORY,FIRST,LAST)

if ~nargin,
    directory = '.';
end

% wd = cd;
% cd(directory);
d = dir(fullfile(directory,'ptv_is.*'));
if nargin < 2
    first = 1;
end
if nargin < 3
    last = length(d);
else
    last = min(last,first + length(d));
end


tmp = textread(fullfile(directory,d(1).name));
numRows = tmp(1);
tmp = tmp(2:end,1:5);

% ind = (tmp(:,1)~=-1 & tmp(:,2)~=-2);
% tmp = tmp(ind,:);

% if ~isempty(tmp)
data(1).prev = tmp(:,1);
data(1).next = tmp(:,2);

data(1).xr = tmp(:,3);
data(1).yr = tmp(:,4);
data(1).zr = tmp(:,5);

[junk,junk,ext,junk] = fileparts(fullfile(directory,d(1).name));
data(1).t = str2double(ext(2:end));
% end

last-first+1
data(last-first+1) = data(1);

for k = first+1:last
    % for i = 1:40000
    i = k - first + 1;
    tmp = textread(fullfile(directory,d(i).name));
    numRows = tmp(1);
    tmp = tmp(2:end,1:5);

    %     ind = (tmp(:,1)~=0 & tmp(:,2)~=-1);
    %     tmp = tmp(ind,:);

    
    data(i).prev = tmp(:,1);
    data(i).next = tmp(:,2);


    data(i).xr = tmp(:,3);
    data(i).yr = tmp(:,4);
    data(i).zr = tmp(:,5);

    data(i).t = data(i-1).t + 1;
    clear tmp
    %
end

File: Python/read_ptv_is_files.py
import os
import glob
import numpy as np

def read_ptv_is_files(directory='.',first=None,last=None):
    fileList = glob.glob(directory+os.sep+'ptv_is.*')
    firstFile = int(fileList[0].split('ptv_is.')[-1])
    if first == None or first < firstFile:
        first = firstFile
	
    lastFile = int(fileList[-1].split('ptv_is.')[-1])
    if last == None or last > lastFile:
	last = lastFile
	
    dt = np.dtype([('prev',np.int),('next',np.int),('x',np.float),('y',np.float),('z',np.float)])
    tmp = np.loadtxt(fileList[0],skiprows=1,dtype=dt)
	
    numRows, = tmp.shape
    numFiles = lastFile - firstFile
	
    desc = np.dtype({'names' : ['data', 't'], 'formats': ['object', 'int']})
    traj = np.empty((numFiles,1),dtype=desc)
    traj[0] = np.array([(tmp,firstFile)],dtype=desc) 
	
    for k in range(1,numFiles):
        tmp = np.loadtxt(fileList[k],skiprows=1,dtype=dt)
        traj[k] = np.array([(tmp,firstFile+k)],dtype=desc) 
	
	return traj
	
	# traj[0]['data'][0]['next']
